python/ray/workflow/workflow_access.py

```diff
@@ -170,16 +195,35 @@ def run_or_resume(
         Returns:
             Workflow execution result that contains the state and output.
         """
-        if workflow_id in self._workflow_outputs and not ignore_existing:
+        if (
+            workflow_id in self._running_workflows
+            or workflow_id in self._queued_workflows
+        ) and not ignore_existing:
             raise RuntimeError(
                 f"The output of workflow[id={workflow_id}] already exists."
             )
+        if (
+            self._max_running_workflows != -1
+            and len(self._running_workflows) >= self._max_running_workflows
+        ):
+            try:
+                self._workflow_queue.put_nowait((job_id, workflow_id))
+                self._queued_workflows.add(workflow_id)
+            except queue.Full:
+                # override with our error message
+                raise queue.Full("Workflow queue has been full") from None
+        else:
+            self._run_or_resume(job_id, workflow_id)
```

Comment:
How about the return type here. Can we still get the result with ray.get ? Is there some py tricks we can use here?